### PR TITLE
Add support for providing credentials when adding chart-repositories

### PR DESCRIPTION
**Pull request by Flydiverny**
**What this PR does / why we need it**:

Add support for providing chart repo credentials when adding private chart repos

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
First time writing go, So be nice <3

### CODE REVIEW

**Comment by Flydiverny** [comment]
Seems like a nicer solution! I pushed a variant with a `helm-repo-extra-args` config.
I did pretty much the same as the first solution to allow for passing different args for different repositories.

so you can do something like this

```sh
ct lint --chart-repos "private-charts=https://charts.private.com,ambassador=https://www.getambassador.io" --helm-repo-extra-args "gleerups-charts=--username user --password secret,ambassador=--no-update"
```

Or yaml config:
```yaml
chart-repos:
  - incubator=https://incubator
  - private=https://private
helm-repo-extra-args:
  - private=--username test --password secret
```

File: pkg/chart/chart.go [hunk 1]
```diff
@@ -197,12 +197,23 @@ func (t *Testing) processCharts(action func(chart string, valuesFiles []string)
 		return nil, errors.Wrap(err, "Error initializing Helm")
 	}
 
+	repoArgs := map[string][]string{}
```
Comment (unguiculus) [inline pkg/chart/chart.go]: I'd say you do too much here. Please use the same logic as for `helm-extra-args` and pass the args into the `Helm` struct.

See https://github.com/helm/chart-testing/blob/master/pkg/chart/chart.go#L163

Comment (Flydiverny) [reply pkg/chart/chart.go]: Not sure I follow, do you agree with the way of providing different args for different repos? And the logic for parsing the args move to NewHelm with AddRepo internally select the correct args when given a repo to add?

Comment (unguiculus) [reply pkg/chart/chart.go]: I guess I should have look more carefully, sorry. I'll have another look at this.

File: app/cmd/root.go [hunk 2]
```diff
@@ -77,6 +77,9 @@ func addCommonLintAndInstallFlags(flags *pflag.FlagSet) {
 	flags.StringSlice("chart-repos", []string{}, heredoc.Doc(`
 		Additional chart repos to add so dependencies can be resolved. May be
 		specified multiple times or separate values with commas`))
+	flags.StringSlice("helm-repo-extra-args", []string{}, heredoc.Doc(`
+		Additional arguments for Helm. (e.g. --username test --password secret)
```
Comment (unguiculus) [inline app/cmd/root.go]: Please state that it's specifically for the `helm repo` command.

File: pkg/chart/chart.go [hunk 3]
```diff
@@ -197,12 +197,23 @@ func (t *Testing) processCharts(action func(chart string, valuesFiles []string)
...
 	for _, repo := range t.config.ChartRepos {
 		repoSlice := strings.SplitN(repo, "=", 2)
 		name := repoSlice[0]
 		url := repoSlice[1]
-		if err := t.helm.AddRepo(name, url); err != nil {
-			return nil, errors.Wrapf(err, "Error adding repo: %s=%s", name, url)
+
+		repoExtraArgs := repoArgs[name]
+		if err := t.helm.AddRepo(name, url, repoExtraArgs); err != nil {
+			return nil, errors.Wrapf(err, "Error adding repo: %s=%s with args %s", name, url, repoExtraArgs)
```
Comment (unguiculus) [inline pkg/chart/chart.go]: Don't include the args in the error message. We don't want to log sensitive data.

File: app/cmd/root.go [hunk 4]
```diff
@@ -77,6 +77,9 @@ func addCommonLintAndInstallFlags(flags *pflag.FlagSet) {
 	flags.StringSlice("chart-repos", []string{}, heredoc.Doc(`
 		Additional chart repos to add so dependencies can be resolved. May be
 		specified multiple times or separate values with commas`))
+	flags.StringSlice("helm-repo-extra-args", []string{}, heredoc.Doc(`
+		Additional arguments for helm repo add. (e.g. --username test --password secret)
+		May be specified multiple times or separate values with commas`))
```
Comment (unguiculus) [inline app/cmd/root.go]: Looking at this again, we should make clear how to configure it, i. e. with `--helm-repo-extra-args <repo>=<args>`. An example in the readme would be great as well.

Comment (unguiculus) [reply app/cmd/root.go]: How about this:

```go
flags.StringSlice("helm-repo-extra-args", []string{}, heredoc.Doc(`
	Additional arguments for the 'helm repo add' command to be 
	specified on a per-repo basis with an equals sign as delimiter 
	(e. g. 'myrepo=--username test --password secret'). May be specified
	multiple times or separate values with commas`))
```

**Comment by Flydiverny** [comment]
@unguiculus pr updated and rebased :)

File: README.md [hunk 5]
```diff
@@ -95,6 +95,23 @@ build-id: pr-42
...
+```yaml
+chart-repos:
+  - incubator=https://incubator.io
+  - basic-auth-repo=https://private.com
+  - ssl-repo=https://self-signed.ca
+helm-repo-extra-args:
+  - ssl-repo=--ca-file ./my-ca.crt
+```
+
+    ct install --config config.yaml ---helm-repo-extra-args "basic-auth=--username user --password secret"
```
Comment (unguiculus) [inline README.md]: One hyphen too many.

```suggestion
    ct install --config config.yaml --helm-repo-extra-args "basic-auth=--username user --password secret"
```

Comment (Flydiverny) [reply README.md]: ops, good catch. pr updated.

**Pull request merged**